runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/DoFnOperator.java

```diff
@@ -1071,7 +1071,11 @@ public void deleteTimer(StateNamespace namespace, String timerId) {
 
     @Override
     public void deleteTimer(StateNamespace namespace, String timerId, TimeDomain timeDomain) {
-      throw new UnsupportedOperationException("Canceling of a timer by ID is not yet supported.");
+      try {
+        cancelPendingTimerById(getContextTimerId(timerId, namespace));
```

Comment:
This is not enough to clean the timer. The method call will only evict the timer meta information used to delete timers by id (Flink does not allow to delete timers by id, you always need to supply the original timestamp of the the timer). See the deleteTimer method below for the additional delete step on Flink's timer service.

Reply:
Have to correct myself, I thought you were calling `cleanupPendingTimer` here, but `cancelPendingTimerById` does delete the timer correctly. Sorry about that.